etc/picongpu/pizdaint-cscs/picongpu.profile.example

```diff
@@ -18,8 +18,8 @@ export MY_NAME="$(whoami) <$MY_MAIL>"
 #
 # if the wrong environment is loaded we switch to the gnu environment
 # note: this loads gcc/5.3.0 (6.0.4 is the version of the programming env!)
-module li 2>&1 | grep "PrgEnv-cray" > /dev/null
-if [ $? -eq 0 ] ; then
+CRAYENV_FOUND=$(module li 2>&1 | grep "PrgEnv-cray" > /dev/null && { echo 0; } || { echo 1; })
```

Comment:
OPTIONAL:
Would it not be more readable if we just execute?:
```bash
module li 2>&1 | grep "PrgEnv-cray" > /dev/null
CRAYENV_FOUND=$?
```

Reply:
the idea here is to avoid depending on `$?` since that means a command needs to fail.

In clean bash scripts, one would set `set -e` at the first executed line, which will prevent that *intermediate failing commands* go unnoticed by a user. Since this script is used in the mighty [*install-all-the-things* script](https://gist.github.com/ax3l/68cb4caa597df3def9b01640959ea56b) you want to see when things fail, e.g. because a URL changed or something did not work as expected.

As an example reference, see: https://github.com/azet/community_bash_style_guide#style-conventions

Reply:
Thanks for the detailed information.